### ISSUE TITLE
[Question] Use of .so files on a Linux environment

### ISSUE THREAD

**Issue by mathieumack** [huysentruitw/SapNwRfc#17]
Hi,
In the readme file, you precised that the sap library (dll or so file) can be copied to the output folder.
Did you already tried it ?
I've tried to publish my console app to an Ubuntu system. So I've copied the .so file provided by SAP, but when I load the package, I got this error :
System.DllNotFoundException: 'Unable to load shared library 'sapnwrfc.so' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libsapnwrfc.so: cannot open shared object file: No such file or directory'
Any idea ?

**Comment by mathieumack** [comment]
Note : I've downloaded source code (not just using the nuget package), and change values for the DllImport attribute in order to load so file, or change also the name of the file, without results.

**Comment by huysentruitw** [comment]
I can confirm that putting the SAP binaries in the folder from which the .NET application gets started works on both Windows and MacOS, haven't tested on Linux but it should work the same. Is it possible to send a screen-shot of your directory layout with the files? Note, when starting an ASP.NET Core project, the folder from which the application gets started is the same as where the .csproj file resides, so behavior is different when compared to a .NET Console application f.e.

**Comment by mathieumack** [comment]
Thanks for the answer.
The .net core application work on a Windows environment with the dll file.
It's only the Linux platform that does not work. I've used the debug on WSL2 configuration (https://devblogs.microsoft.com/dotnet/debug-your-net-core-apps-in-wsl-2-with-visual-studio/) to debug my app on Ubuntu.
I will check my app directly published on Azure and update this thread with results and solution if I found it.
